crds/workspace.devfile.io_devworkspaces.yaml

```diff
@@ -6620,6 +6983,126 @@ spec:
+                                                  devfileRegistryUrl:
+                                                    description: Devfile Registry
+                                                      URL to pull the Dockerfile from
+                                                      when using the Devfile Registry
+                                                      as Dockerfile src. To ensure
+                                                      the Dockerfile gets resolved
+                                                      consistently in different environments,
+                                                      it is recommended to always
+                                                      specify the `devfileRegistryUrl`
+                                                      when `Id` is used.
+                                                    type: string
+                                                  id:
+                                                    description: Id in a devfile registry
+                                                      that contains a Dockerfile.
+                                                      The src in the OCI registry
+                                                      required for the Dockerfile
+                                                      build will be downloaded for
+                                                      building the image.
+                                                    type: string
+                                                type: object
+                                              rootRequired:
+                                                description: "Specify if a privileged
+                                                  builder pod is required. \n Default
+                                                  value is `false`"
+                                                type: boolean
+                                              uri:
+                                                description: URI Reference of a Dockerfile.
+                                                  It can be a full URL or a relative
+                                                  URI from the current devfile as
+                                                  the base URI.
+                                                type: string
+                                            type: object
+                                          imageName:
+                                            description: Name of the image for the
+                                              resulting outerloop build
+                                            type: string
+                                          imageType:
+                                            description: Type of image
+                                            enum:
+                                            - Dockerfile
```

Comment:
it's not really clear for me... What are possible alternatives?

Reply:
Its outlined in the proposal. Dockerfile is one type of image component. Rest will be s2i, tekton, etc.

via https://github.com/devfile/api/blob/main/docs/proposals/devfile/outer-loop-build-and-deploy.md#build-the-image